index.js

```diff
@@ -327,71 +318,131 @@ if(!helpers.isTestEnvironment()){
 }
 
 // Handler for income SmartThings POST requests
-app.post('/api/st', function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
-    smartapp.handleHttpCallback(req, res);
+app.post('/api/st', Validator.body(['lifecycle']).exists(), function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
+    try {
+        const validationErrors = Validator.validationResult(req)
+        if(validationErrors.isEmpty()){
+            smartapp.handleHttpCallback(req, res)
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
+        }
+    } catch (err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
 });
 
 // Handler for income XeThru POST requests
-app.post('/api/xethru', async (req, res) => {
-    // eslint-disable-next-line no-unused-vars -- might be useful in the future to know what all we have access to in the body
-    const {deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s} = req.body;
-
-    redis.addXeThruSensorData(req, res);
-    handleSensorRequest(locationid);
-});
-
-app.post('/api/doorTest', async(req, res) => {
-    const {locationid} = req.body;
-    await redis.addDoorTestSensorData(req, res);
-    await handleSensorRequest(locationid)
+app.post('/api/xethru', Validator.body(['locationid', 'state', 'rpm', 'distance', 'mov_f', 'mov_s']).exists(), async (req, res) => {
+    try {
+        const validationErrors = Validator.validationResult(req)
+
+        if(validationErrors.isEmpty()){
+            // eslint-disable-next-line no-unused-vars -- might be useful in the future to know what all we have access to in the body
+            const { deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s } = req.body;
+
+            await redis.addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s);
+            await handleSensorRequest(locationid);
+            res.status(200).send()
```

Comment:
Looks like there used to be a `.json("OK")` instead of `send()` for this response before. It probably doesn't matter, but to be on the safe side, you could add it back in.

Reply:
In addition to send or as a replacement? Can do either way :)

Reply:
It looks like our other functions are using just `json("OK")`. I guess it doesn't need a `send()` if you do that :shrug:

Reply:
Okay sounds good haha

Reply:
Oh, now that I'm looking at this in my text editor I remember why I did this, it was for consistency since the other things for 400 and 500 are formatted this way in the else & catch

```
} else {
  helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
  res.status(400).send()
  }
} catch (err) {
   helpers.log(err)
   res.status(500).send()
}
```

Reply:
That also makes sense. I'd just say that taking out the `.json("OK")` will require more testing because you've changed the conversation between the devices. You're choice :smile:

Reply:
Yeah I will probably just put it back for now just on the off chance 😅